examples/databases/src/sqlx.rs

```diff
@@ -42,65 +47,47 @@ async fn list(db: &State<Db>) -> Result<Json<Vec<i64>>> {
 }
 
 #[get("/<id>")]
-async fn read(db: &State<Db>, id: i64) -> Option<Json<Post>> {
+async fn read(mut db: Connection, id: i64) -> Option<Json<Post>> {
     sqlx::query!("SELECT id, title, text FROM posts WHERE id = ?", id)
-        .fetch_one(&**db)
+        .fetch_one(&mut *db)
         .map_ok(|r| Json(Post { id: Some(r.id), title: r.title, text: r.text }))
         .await
         .ok()
 }
 
 #[delete("/<id>")]
-async fn delete(db: &State<Db>, id: i64) -> Result<Option<()>> {
+async fn delete(mut db: Connection, id: i64) -> Result<Option<()>> {
     let result = sqlx::query!("DELETE FROM posts WHERE id = ?", id)
-        .execute(&**db)
+        .execute(&mut *db)
         .await?;
 
     Ok((result.rows_affected() == 1).then(|| ()))
 }
 
 #[delete("/")]
-async fn destroy(db: &State<Db>) -> Result<()> {
-    sqlx::query!("DELETE FROM posts").execute(&**db).await?;
+async fn destroy(mut db: Connection) -> Result<()> {
+    sqlx::query!("DELETE FROM posts").execute(&mut *db).await?;
 
     Ok(())
 }
 
 async fn init_db(rocket: Rocket<Build>) -> fairing::Result {
-    use rocket_sync_db_pools::Config;
-
-    let config = match Config::from("sqlx", &rocket) {
-        Ok(config) => config,
-        Err(e) => {
-            error!("Failed to read SQLx config: {}", e);
-            return Err(rocket);
+    match rocket.state::<Db>() {
```

Comment:
This feels like an abstraction has just leaked. I think we want something like https://api.rocket.rs/v0.5-rc/rocket/struct.State.html#method.get. Maybe `Db::pool(&rocket);` and `Db::conn(&rocket).await;`.

Reply:
Leaking the abstraction in this particular way (putting the derived database type directly in managed state) was intentional, because I thought it was much simpler to understand. `Db::pool` and `Db::conn` already take `&self`, so this would logically be adding `Db::pool_from_rocket(&rocket)` and `Db::conn_from_rocket(&rocket).await`. Happy to make that change, though.

Reply:
The problem is that this presents two different interfaces: an abstracted `Db::fairing()` and now a `Db` in managed state. If the user had instead been asked to do `.manage(Db::init()?)` or something to that effect, this would be fine. But otherwise, `.attach(Db::fairing())` says nothing about something going into managed state.